common/cgroup/cgroup.go

```diff
@@ -69,32 +79,61 @@ func MaybeAddIsolator(opts []*unit.UnitOption, isolator string, limit *resource.
 	if limit == nil {
 		return opts, nil
 	}
-	if IsIsolatorSupported(isolator) {
+	isSupported, err := IsIsolatorSupported(isolator)
+	if err != nil {
+		return nil, err
+	}
+
+	if isSupported {
 		opts, err = isolatorFuncs[isolator](opts, limit)
-		if err != nil {
-			return nil, err
-		}
 	} else {
 		fmt.Fprintf(os.Stderr, "warning: resource/%s isolator set but support disabled in the kernel, skipping\n", isolator)
 	}
+
+	if err != nil {
+		return nil, err
+	}
 	return opts, nil
 }
 
 // IsIsolatorSupported returns whether an isolator is supported in the kernel
-func IsIsolatorSupported(isolator string) bool {
-	if files, ok := cgroupControllerRWFiles[isolator]; ok {
-		for _, f := range files {
-			isolatorPath := filepath.Join("/sys/fs/cgroup/", isolator, f)
-			if _, err := os.Stat(isolatorPath); os.IsNotExist(err) {
-				return false
-			}
+func IsIsolatorSupported(isolator string) (bool, error) {
```

Comment:
The legacy check was here because sometimes, even if a controller is enabled, the knobs we need to touch are not available. In cgroup v2 I think we can assume that if a controller is enabled, the related isolator is supported.